src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -157,6 +157,161 @@ private[spark] object Config {
   private def k8sLabelListValidator(labelList: String): Boolean = {
     parseCommaSeparatedK8sLabels(labelList).forall(_.isSuccess)
   }
+
+  val ARMADA_QUEUE: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.queue")
+      .doc("The Armada queue that will be used for running the driver and executor pods.")
+      .version("1.0.0")
+      .stringConf
+      .createWithDefault("default")
+
+  val DRIVER_JOB_SET_ID: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.driver.job.set.id")
+      .doc("The Armada job set id that will be used for running the driver pods.")
+      .version("1.0.0")
+      .stringConf
+      .createWithDefault("Driver-Job-Set")
+
+  val EXECUTOR_JOB_SET_ID: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.executor.job.set.id")
+      .doc("The Armada job set id that will be used for running the executor pods.")
+      .version("1.0.0")
+      .stringConf
+      .createWithDefault("Executor-Job-Set")
+
+  val ARMADA_NAMESPACE: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.namespace")
+      .doc("The namespace that will be used for running the driver and executor pods.")
+      .version("1.0.0")
+      .stringConf
+      .createWithDefault("default")
+
+  val ARMADA_REMOTE_MASTER: OptionalConfigEntry[String] =
+    ConfigBuilder("spark.armada.remote.master")
```

Comment:
What is the difference to the URL given in `--master`?

Reply:
It is because within the armada cluster the armada master may have a different url than outside.  Right now, in the submit script, we use:
```
ARMADA_MASTER=armada://localhost:30002
```

But this is the remote master in the script:
```
    --conf spark.armada.remote.master=local://armada://armada-server.armada.svc.cluster.local:50051 \
```

Reply:
I think `armada://host:port` is the form we ultimately want to work and `local://armada://host:port` is a temporary workaround to make Spark code work with an `armada://` URL scheme.

I'd say we prefix the Spark master with`localhost://` if it starts with `armada://` when we set `--master` for the driver pod. That is internal knowledge and users shouldn't care.

https://github.com/armadaproject/armada-spark/blob/6a95494f5e3c72848638022f8c71262b9ca103de/src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala#L421-L422

Reply:
The problem is the host:port, it may not be the same internally vs externally:
 "armada-server.armada.svc.cluster.local:50051", 
vs "localhost:30002"